File: src/main/csharp/Data/EntityRepository.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Data.SqlClient;
using App.Models;

namespace App.Data
{
    public class EntityRepository
    {
        private readonly SqlConnection _connection;
        
        public EntityRepository(SqlConnection connection)
        {
            _connection = connection;
        }
        
        public Entity FindById(long id)
        {
            using (var cmd = new SqlCommand("SELECT Id, Name, Email, Status, Active, Balance FROM Entities WHERE Id = @id", _connection))
            {
                cmd.Parameters.AddWithValue("@id", id);
                using (var reader = cmd.ExecuteReader())
                {
                    if (reader.Read())
                    {
                        return new Entity
                        {
                            Id = reader.GetInt64(0),
                            Name = reader.GetString(1),
                            Email = reader.GetString(2),
                            Status = reader.GetInt32(3),
                            Active = reader.GetBoolean(4),
                            Balance = reader.GetDouble(5)
                        };
                    }
                }
            }
            return null;
        }
        
        public List<Entity> FindAll()
        {
            var entities = new List<Entity>();
            using (var cmd = new SqlCommand("SELECT Id, Name, Email, Status, Active, Balance FROM Entities", _connection))
            {
                using (var reader = cmd.ExecuteReader())
                {
                    while (reader.Read())
                    {
                        entities.Add(new Entity
                        {
                            Id = reader.GetInt64(0),
                            Name = reader.GetString(1),
                            Email = reader.GetString(2),
                            Status = reader.GetInt32(3),
                            Active = reader.GetBoolean(4),
                            Balance = reader.GetDouble(5)
                        });
                    }
                }
            }
            return entities;
        }
        
        public List<Entity> FindByStatus(int status)
        {
            var entities = new List<Entity>();
            using (var cmd = new SqlCommand("SELECT Id, Name, Email, Status, Active, Balance FROM Entities WHERE Status = @status", _connection))
            {
                cmd.Parameters.AddWithValue("@status", status);
                using (var reader = cmd.ExecuteReader())
                {
                    while (reader.Read())
                    {
                        entities.Add(new Entity
                        {
                            Id = reader.GetInt64(0),
                            Name = reader.GetString(1),
                            Email = reader.GetString(2),
                            Status = reader.GetInt32(3),
                            Active = reader.GetBoolean(4),
                            Balance = reader.GetDouble(5)
                        });
                    }
                }
            }
            return entities;
        }
    }
}

File: src/main/java/com/app/repository/EntityRepository.java
package com.app.repository;

import com.app.model.Entity;
import java.sql.*;
import java.util.ArrayList;
import java.util.List;
import java.util.Optional;

public class EntityRepository {
    
    private Connection connection;
    
    public EntityRepository(Connection connection) {
        this.connection = connection;
    }
    
    public Optional<Entity> findById(long id) throws SQLException {
        String sql = "SELECT id, name, email, status, active, balance FROM entities WHERE id = ?";
        try (PreparedStatement stmt = connection.prepareStatement(sql)) {
            stmt.setLong(1, id);
            ResultSet rs = stmt.executeQuery();
            if (rs.next()) {
                Entity e = new Entity();
                e.setId(rs.getLong("id"));
                e.setName(rs.getString("name"));
                e.setEmail(rs.getString("email"));
                e.setStatus(rs.getInt("status"));
                e.setActive(rs.getBoolean("active"));
                e.setBalance(rs.getDouble("balance"));
                return Optional.of(e);
            }
        }
        return Optional.empty();
    }
    
    public List<Entity> findAll() throws SQLException {
        List<Entity> entities = new ArrayList<>();
        String sql = "SELECT id, name, email, status, active, balance FROM entities";
        try (Statement stmt = connection.createStatement()) {
            ResultSet rs = stmt.executeQuery(sql);
            while (rs.next()) {
                Entity e = new Entity();
                e.setId(rs.getLong("id"));
                e.setName(rs.getString("name"));
                e.setEmail(rs.getString("email"));
                e.setStatus(rs.getInt("status"));
                e.setActive(rs.getBoolean("active"));
                e.setBalance(rs.getDouble("balance"));
                entities.add(e);
            }
        }
        return entities;
    }
    
    public List<Entity> findByStatus(int status) throws SQLException {
        List<Entity> entities = new ArrayList<>();
        String sql = "SELECT id, name, email, status, active, balance FROM entities WHERE status = ?";
        try (PreparedStatement stmt = connection.prepareStatement(sql)) {
            stmt.setInt(1, status);
            ResultSet rs = stmt.executeQuery();
            while (rs.next()) {
                Entity e = new Entity();
                e.setId(rs.getLong("id"));
                e.setName(rs.getString("name"));
                e.setEmail(rs.getString("email"));
                e.setStatus(rs.getInt("status"));
                e.setActive(rs.getBoolean("active"));
                e.setBalance(rs.getDouble("balance"));
                entities.add(e);
            }
        }
        return entities;
    }
}

